share/completions/konsole.fish

```diff
@@ -1,35 +1,36 @@
-# Completion for `konsole
+# Completion for konsole
 
-complete -c konsole -l help -d 'Show help and exit'
-complete -c konsole -s v -d 'Show version and exit'
-complete -c konsole -l version -d 'Show version and exit'
-complete -c konsole -l help-qt -d 'Show Qt specific options and exit'
-complete -c konsole -l help-kde -d 'Show KDE specific options and exit'
+complete -c konsole -f
+
+complete -c konsole -l help -xs h -d 'Show help and exit'
 complete -c konsole -l help-all -d 'Show all options and exit'
-complete -c konsole -l author -d 'Show author information and exit'
-complete -c konsole -l license -d 'Show license information and exit'
-complete -c konsole -l name -d 'Window class'
-complete -c konsole -l ls -d 'Login shell'
-complete -c konsole -s T -d 'Window name'
-complete -c konsole -l tn -d 'Terminal type'
-complete -c konsole -l noclose -d 'Do not close when command exists'
-complete -c konsole -l nohist -d 'Do not save history'
-complete -c konsole -l nomenubar -d 'Hide menu bar'
-complete -c konsole -l notabbar -d 'Hide tab bar'
-complete -c konsole -l notoolbar -d 'Hide tab bar'
-complete -c konsole -l noframe -d 'Hide frame bar'
-complete -c konsole -l noscrollbar -d 'Hide scrollbar'
-complete -c konsole -l noxft -d 'Do not use atni-aliasing'
-complete -c konsole -l noresize -d 'Do not allow window to be resized'
-complete -c konsole -l vt_sz -d 'Window size'
-complete -c konsole -l type -d 'Session type'
-complete -c konsole -l types -d 'List session types'
-complete -c konsole -l keytab -d 'Keytab name'
-complete -c konsole -l keytabs -d 'List keytabs'
-complete -c konsole -l profile -d Profile
-complete -c konsole -l profiles -d 'List profiles'
-complete -c konsole -l schema -d Schema
-complete -c konsole -l schemas -d 'List schemas'
-complete -c konsole -l script -d 'Enable extended DCOP Qt functions'
-complete -c konsole -l workdir -d 'Working directory'
-complete -c konsole -s e -d 'Command to execute'
+complete -c konsole -l version -xs v -d 'Show version and exit'
+complete -c konsole -l author -d 'Show author info and exit'
+complete -c konsole -l license -d 'Show license info and exit'
+complete -c konsole -l desktopfile -d 'Base file name of desktop entry'
+
+set -l profiles (konsole --list-profiles)
+complete -c konsole -l profile -d 'Name of profile to use'
+for p in $profiles
+    complete -c konsole -n "__fish_seen_subcommand_from --profile; and not __fish_seen_subcommand_from $profiles" -a $p
+end
```

Comment:
Instead of calling `konsole --list-profiles` when this completion is sourced (bad, hangs shell for some ms), this should be dynamic. No need for `__fish_seen_subcommand_from --profile` and `not __fish_seen_subcommand_from $profiles` because, if I understand it correctly, this is just a single mandatory argument to the long option `--profile`, so just complete it as a long argument `complete -c konsole -l profile -rxa (konsole --list-profiles)`